src/components/PopupBase/__tests__/PopupBase.spec.tsx

```diff
@@ -44,119 +50,127 @@ describe('<PopupBase> that adds basic anchor/popup functionality to rendered com
         });
-        it('should close popup when requested', () => {
+
+        it('should close popup when requested', async () => {
+            const user = userEvent.setup();
             // trigger setPopupVisibility(true) through our dummy component
-            const triggerButton = wrapper.find('button');
-            triggerButton.simulate('click');
-            expect(wrapper.find('Popover')).toHaveLength(1);
+            const triggerButton = screen.getByRole('button');
+            await user.click(triggerButton);
 
-            triggerButton.simulate('click');
-            expect(wrapper.find('Popover')).toHaveLength(0);
+            expect(screen.queryByRole('group')).toBeInTheDocument();
+
+            await user.click(triggerButton);
+
+            expect(screen.queryByRole('group')).not.toBeInTheDocument();
         });
     });
 
     describe('click and keydown event handling', () => {
-        let togglePopup;
         const onCloseMock = jest.fn();
-
-        const clickDocument = useDocumentEvent('click');
-        const keydownDocument = useDocumentEvent('keydown');
+        const togglePopup = async (user) => {
+            await user.click(screen.getAllByRole('button')[0]);
+        };
 
         beforeEach(() => {
-            wrapper = mount(
+            view = render(
                 <PopupBase
                     anchorRenderer={anchorRendererMock}
                     popupRenderer={popupRendererMock}
                     onClose={onCloseMock}
                 />
             );
+        });
+
+        it('should close open popup if outside is clicked', async () => {
+            const user = userEvent.setup();
+            await togglePopup(user);
+
+            expect(screen.queryByRole('group')).toBeInTheDocument();
+
+            await togglePopup(user);
 
-            togglePopup = () => {
-                wrapper.find('button').at(0).simulate('click');
-            };
+            expect(screen.queryByRole('group')).not.toBeInTheDocument();
         });
 
-        afterEach(() => {
-            onCloseMock.mockReset();
+        it('should call onClose if outside is clicked', async () => {
+            const user = userEvent.setup();
+            await togglePopup(user);
+            await user.click(document.body);
+
+            expect(onCloseMock).toHaveBeenCalled();
         });
 
-        it('should close open popup if outside is clicked', () => {
-            togglePopup();
-            expect(wrapper.find('Popover')).toHaveLength(1);
+        it('should close open popup on Escape press', async () => {
+            const user = userEvent.setup();
+            await togglePopup(user);
 
-            // @ts-ignore
-            clickDocument();
-            wrapper.update();
+            expect(screen.getByRole('group')).toBeInTheDocument();
 
-            expect(wrapper.find('Popover')).toHaveLength(0);
+            await user.keyboard(`[${ESCAPE_KEY}]`);
```

Comment:
I think it is more confusing to use the ESCAPE_KEY variable than simply use [Escape], what do you think?

Reply:
For me using constant is better.

Reply:
Also test cases contain this approach before

Reply:
I'm fine with it.